### DISCUSSION TITLE
Outbound Firewall Rules Failing

### DISCUSSION THREAD

**Discussion by jmcintyre** [wixtoolset/issues#7774] [Questions]
I don't seem to have any problems installing inbound firewall exceptions, but outbound exceptions are always failing for me. Here is a very simple version where just switching Outbound to "yes" causes the installation to fail.

```xml
  <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="Binary" Guid="*">
        <File Id="BinaryName" KeyPath="yes" Source="my.exe">
          <fire:FirewallException Name="ExceptionName" Scope="any" Outbound="yes"/>
        </File>
      </Component>
    </DirectoryRef>
```

The logs look something like this (edited down to match the above example):

```
ExecFirewallExceptions:  Installing firewall exception2 ExceptionName (C:\Program Files\Company\my.exe)
ExecFirewallExceptions:  Error 0x80070057: failed to add app to the authorized apps list
ExecFirewallExceptions:  Error 0x80070057: failed to add/update application exception for name 'ExceptionName', file 'C:\Program Files\Company\my.exe'
CustomAction Wix4ExecFirewallExceptionsInstall_X64 returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
Action ended 18:39:56: InstallFinalize. Return value 3.
```

**Comment by iohann95** [comment]
I'm also having this problem, can't figure out how to solve it.

**Comment by Thylon1** [comment]
If you want to specify an Outbound firewall rule you should/can not assign it to a file.
You will also need to specify a port on which to allow incoming requests.

So your code should looks like this:
```
  <DirectoryRef Id="INSTALLFOLDER">
      <Component Id="Binary" Guid="*">
        <File Id="BinaryName" KeyPath="yes" Source="my.exe" />
        <fire:FirewallException Name="ExceptionName" Scope="any" Outbound="yes" Port="5000" />
      </Component>
    </DirectoryRef>
```

You can also find another example in the source code [here](https://github.com/wixtoolset/Firewall.wixext/blob/master/src/test/WixToolsetTest.Firewall/TestData/UsingOutboundFirewall/PackageComponents.wxs).

Unfortunately, I am not sure if you can create an outbound firewall rule that is associated with a file.
